src/OnboardingSPA/components/NewfoldInterfaceSkeleton/SiteGen/index.js

```diff
@@ -82,10 +83,14 @@ const SiteGen = () => {
 							retryCount + 1
 						);
 					}
+					if ( retryCount === MAX_RETRIES_SITE_GEN ) {
```

Comment:
Do we need to include this `if` check? From the logic, it seems like we can skip this and simply update the error state to true.

Reply:
it's a recursive function, in some cases it's not able to catch the error. Hence the check.

Reply:
We need to add a `return` then instead of `if`

Reply:
Done